### PR TITLE
feat: filter flagged duplicate sets

### PR DESCRIPTION
**Pull request by emilyjablonski**
# Pull Request Template

## Issue Overview

This PR addresses #4100

- [x] This change addresses the issue in full

## Description

Allows for the ability to search on application flagged sets by the first/last names of any applicant in a set.

Pending Set
The information in the first column was duplicated in the next two columns. Now the duplicate information is removed and we are showing the names of all applicants in the first column since there really is no "primary applicant" in a set and it was really confusing to search for an applicant name and have a line appear that didn't show that applicant's name, but because the applicant was in the set. 
Before:
Screenshot 2024-05-21 at 1 11 28 PM
After:
Screenshot 2024-05-21 at 1 11 33 PM

Resolved Set
Similar changes, duplicate information is removed and all applicant names are being shown
Before:
Screenshot 2024-05-21 at 1 12 00 PM
After:
Screenshot 2024-05-21 at 1 12 14 PM

Additionally, because that first column now needs to take up lots of space, I had to make adjustments to the sizes of the columns. I saw that the column sizes were wonky on all partners tables, so I lightly adjusted the spacings based on what I learned in the flagged set table.

## How Can This Be Tested/Reviewed?

Go to a listing with pending and resolved sets of application duplicates and try to search for names.

You can also look at the listing and application tables if you want to see some new column sizes. It's fairly subtle there.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: sites/partners/page_content/locale_overrides/general.json [hunk 1]
```diff
@@ -33,7 +33,7 @@
   "application.details.householdSize": "Household Size",
   "application.details.language": "Application Language",
   "application.details.monthlyIncome": "Monthly Income",
-  "application.details.number": "Application Number",
+  "application.details.number": "Confirmation Code",
```
Comment (emilyjablonski) [inline sites/partners/page_content/locale_overrides/general.json:36]: We were calling the confirmation code "Application Number" which I worried was confusing especially given the latest lottery

File: sites/partners/src/components/applications/ApplicationsColDefs.ts [hunk 2]
```diff
@@ -263,19 +260,19 @@ export function getColDefs(maxHouseholdSize: number, countyCode: string) {
...
-      minWidth: 100,
+      minWidth: 50,
       valueFormatter: ({ value }) => {
         if (!value) return ""
         return value ? value : t("t.none")
       },
     },
     {
-      headerName: t("applications.table.additionalPhoneType"),
+      headerName: t("applications.table.additionalPhoneTypeShortened"),
```
Comment (emilyjablonski) [inline sites/partners/src/components/applications/ApplicationsColDefs.ts:270]: Just made this shorter, it was a v long string for a table header

File: sites/partners/page_content/locale_overrides/general.json (hunk 1 above)
Comment (YazeedLoonat) [reply sites/partners/page_content/locale_overrides/general.json:36]: I think we may want to check in with PM/design (if you hadn't already) about this change

File: sites/partners/src/components/applications/PaperApplicationDetails/sections/DetailsApplicationData.tsx [hunk 3]
```diff
@@ -18,7 +18,7 @@ const DetailsApplicationData = () => {
     <SectionWithGrid heading={t("application.details.applicationData")} inset>
       <Grid.Row>
         <FieldValue label={t("application.details.number")} testId="number">
-          {application.confirmationCode || application.id}
+          {application.confirmationCode ? application.confirmationCode : t("t.n/a")}
```
Comment (YazeedLoonat) [inline sites/partners/src/components/applications/PaperApplicationDetails/sections/DetailsApplicationData.tsx:21]:
```suggestion
          {application.confirmationCode ??  t("t.n/a")}
```

File: sites/partners/src/pages/application/[id]/applicationsCols.tsx [hunk 4]
```diff
@@ -18,13 +18,14 @@ export const getCols = () => [
       if (!data?.id && !data?.confirmationCode) return ""
       return (
         <span className={"text-blue-900"}>
-          <Link href={`/application/${data.id}`}>{data.confirmationCode || data.id}</Link>
+          <Link href={`/application/${data.id}`}>
+            {data.confirmationCode ? data.confirmationCode : t("t.n/a")}
```
Comment (YazeedLoonat) [inline sites/partners/src/pages/application/[id]/applicationsCols.tsx:22]:
```suggestion
            {data.confirmationCode ??  t("t.n/a")}
```

File: sites/partners/src/pages/listings/[id]/applications/pending/index.tsx [hunk 5]
```diff
@@ -51,27 +52,23 @@ const ApplicationsList = () => {
       cellRenderer: "formatLinkCell",
       valueGetter: ({ data }) => {
         if (!data?.applications?.length) return ""
-        const applicant = data.applications[0]?.applicant
 
-        return `${applicant.firstName} ${applicant.lastName}: ${data.rule}`
+        const applications = data.applications.reduce((acc, curr) => {
```
Comment (YazeedLoonat) [inline sites/partners/src/pages/listings/[id]/applications/pending/index.tsx:56]: I don't think we need to address this here/in this pr but I'm curious on your thoughts on pulling stuff like this out into helper files so they are more easily unit testable

File: sites/partners/src/pages/listings/[id]/applications/resolved/index.tsx [hunk 6]
```diff
@@ -36,22 +37,21 @@ const ApplicationsList = () => {
       cellRenderer: "formatLinkCell",
       valueGetter: ({ data }) => {
         if (!data?.applications?.length) return ""
-        const applicant = data.applications[0]?.applicant
+        const applications = data.applications.reduce((acc, curr) => {
```
Comment (YazeedLoonat) [inline sites/partners/src/pages/listings/[id]/applications/resolved/index.tsx:40]: with the previous comment, if this was a helper we can leverage it here as well

**Review by YazeedLoonat — COMMENTED** [review]
Looks good, a few nits and some questions

File: sites/partners/page_content/locale_overrides/general.json (hunk 1 above)
Comment (emilyjablonski) [reply sites/partners/page_content/locale_overrides/general.json:36]: Yep yep scheduled a message for Em tmrw :)

File: sites/partners/src/pages/listings/[id]/applications/pending/index.tsx (hunk 5 above)
Comment (emilyjablonski) [reply sites/partners/src/pages/listings/[id]/applications/pending/index.tsx]: did both! 🫡

**Review by YazeedLoonat — APPROVED** [review]
LGTM

**Pull request merged**